setup/src/installer.ts

```diff
@@ -116,6 +116,7 @@ async function isInstalled(
       .catch(() => undefined);
 
     if (installedPath) {
+      await exec(await ghcupBin(os), ['install', tool, version]);
```

Comment:
Rather than installing cabal in the "check if this is installed" function, I think it makes more sense to take advantage of ghcup now installing cabal as `~/.ghcup/bin/cabal-$version` and use that to check for its existence. Does that also sound workable to you?

Then the isInstalled logic would actually work and the rest of the pipeline (including emitting warnings, actually installing cabal if needed, etc) would continue as normal and we avoid duplicating install logic.

```diff
async function isInstalled(
     const installedPath = await fs
-      .access(`${ghcupPath}/cabal`)
+      .access(`${ghcupPath}/cabal-${version}`)
       .then(() => ghcupPath)
       .catch(() => undefined);
```

Reply:
Your suggestion works and have been pushed.